control/metrics.go

```diff
@@ -289,7 +301,7 @@ func (mc *metricCatalog) AddLoadedMetricType(lp *loadedPlugin, mt core.Metric) e
 		return err
 	}
 	newMt := metricType{
-		Plugin:             lp,
+		Plugin:             *lp,
```

Comment:
That was my initial implementation, but I thought it's too verbose and changed it to dereferencing `loadedPlugin`. I guess it was mistake and I missed the need of deep copy.